R/visualization.R

```diff
@@ -0,0 +1,222 @@
+  if (!inherits(drug_response, "drug.reponse")) {
+    stop("Input must be a drug.reponse object.")
+  }
+
+  # Define periods from the response object for consistency
+  before_period_def <- drug_response$lab_response_period$before_period
+  after_period_def <- drug_response$lab_response_period$after_period
+
+  lab_data_periods <- drug_response$all_measurements %>%
+    filter(!is.na(.data$first_drug_age) & !is.na(.data$MEASUREMENT_VALUE_HARMONIZED)) %>%
+    mutate(period = case_when(
+      between(.data$time_to_drug, before_period_def[1], before_period_def[2]) ~ "Before",
+      between(.data$time_to_drug, after_period_def[1], after_period_def[2]) ~ "After",
+      TRUE ~ NA_character_
+    )) %>%
+    filter(!is.na(.data$period))
+
+  plot_data <- lab_data_periods
+  if (remove_outliers) {
+    plot_data <- plot_data %>%
+      group_by(.data$first_drug, .data$period) %>%
+      mutate(
+        Q1 = quantile(.data$MEASUREMENT_VALUE_HARMONIZED, 0.25, na.rm = TRUE),
+        Q3 = quantile(.data$MEASUREMENT_VALUE_HARMONIZED, 0.75, na.rm = TRUE),
+        IQR = .data$Q3 - .data$Q1
+      ) %>%
+      filter(
+        .data$MEASUREMENT_VALUE_HARMONIZED >= (.data$Q1 - 1.5 * .data$IQR) &
+        .data$MEASUREMENT_VALUE_HARMONIZED <= (.data$Q3 + 1.5 * .data$IQR)
+      ) %>%
+      ungroup()
+  }
+
+  # Generate plot
+  p <- ggplot(plot_data,
+              aes(x = .data$period, y = .data$MEASUREMENT_VALUE_HARMONIZED,
+                  fill = .data$period)) +
+    ggplot2::geom_boxplot(outlier.shape = if (remove_outliers) NA else 19) +
+    ggpubr::stat_compare_means(method = "t.test",
+                               label = "p.format", paired = FALSE,
+                               label.x = 1.5, label.y.npc = 0.9) +
+    labs(
+      title = "Distribution of Lab Values Before and After First Drug Purchase",
+      x = "Period Relative to Drug Purchase",
+      y = "Harmonised Measurement Value"
+    ) +
+    theme_minimal() +
+    facet_wrap(~.data$first_drug, scales = "free_y")
+
+  return(p)
+}
+
+
+#' @title Generate an UpSet plot of drug purchase combinations
+#' @description Creates and saves an UpSet plot to visualize the overlap of purchased drug ATC codes.
+#' @param drug_response A `drug.response` object created by `create_drug_response`.
+#' @param out_file_prefix A string to use as the prefix for the output PDF file.
+#' @return NULL
+#' @export
+summarize_drug_purchases_upset <- function(drug_response, out_file_prefix) {
+  if (!inherits(drug_response, "drug.reponse")) {
+    stop("Input must be a drug.reponse object.")
+  }
+
+  upset_data <- drug_response$all_drug_purchases %>%
+    select("FINNGENID", "ATC") %>%
+    distinct() %>%
+    mutate(value = 1) %>%
+    tidyr::pivot_wider(names_from = "ATC", values_from = "value", values_fill = 0) %>%
+    select(-"FINNGENID") %>%
+    as.data.frame()
+
+  pdf(paste0(out_file_prefix, "_upset_plot.pdf"), width = 10, height = 7)
+
+  UpSetR::upset(upset_data,
```

Comment:
seems not to plot anything in into pdf unless explicitly callint print on the returned object in Radian terminal. 
Add explicit print